src/makeInfinityAdder.js

```diff
@@ -6,7 +6,24 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+  let emptyCall = 0;
+
+  const adder = (num) => {
+    if (typeof num === 'undefined') {
+      sum = emptyCall > 0
+        ? 0
+        : sum;
+      emptyCall++;
```

Comment:
It looks like an unnecessary complication in this situation. Furthermore, even though it may pass the tests, it does not work as expected. To illustrate, try testing your function in the following scenario:
``` js
const myAdder = makeInfinityAdder();

myAdder(5)(5)(5);
myAdder(4);
console.log(myAdder()); // Expected: 19. Received: 19
myAdder(5)(5)(5);
myAdder(4);
console.log(myAdder()); // Expected: 19. Received: 0
```
Please ensure that you carefully read the task. Once you verify that the function is called without any parameters, you should clear the sum variable and return the cleared value (using a temporary variable may be helpful here).